frontend/src/components/ConfigForm.vue

```diff
@@ -43,67 +79,120 @@
+          :mandatory="true"
+        >
+          <v-radio
+            label="Manual commit to Bitbucket (deploy with Jenkins)"
+            value="deploymentManual"
+          ></v-radio>
+          <p v-if="userAppCfg.deploymentMethod === 'deploymentManual'" class="underRadioField">
+            <a
+              href="https://github.com/bcgov/nr-get-token/wiki/WebADE-Access"
+              target="_blank"
+            >Instructions for manual deployment</a>
+          </p>
+          <v-radio label="Direct Deploy" value="deploymentDirect"></v-radio>
+
+          <v-layout row wrap>
+            <v-flex xs12>
+              <v-text-field
+                v-if="userAppCfg.deploymentMethod === 'deploymentDirect'"
+                label="Password"
+                required
+                :value="userAppCfg.userEnteredPassword"
+                v-on:keyup.stop="updateAppCfgField('userEnteredPassword', $event.target.value)"
+                class="underRadioField"
+                :rules="passwordRules"
+                :append-icon="showPw ? 'visibility' : 'visibility_off'"
+                :type="showPw ? 'text' : 'password'"
+                hint="At least 8 characters"
+                :counter="60"
+                @click:append="showPw = !showPw"
+              ></v-text-field>
+            </v-flex>
+          </v-layout>
+        </v-radio-group>
 
-      <v-btn
-        v-if="userAppCfg.deploymentMethod === 'deploymentDirect'"
-        color="success"
-        @click="submitConfig"
-      >Submit</v-btn>
-      <v-btn flat @click="appConfigStep = 1">Back</v-btn>
+        <v-btn
+          v-if="userAppCfg.deploymentMethod === 'deploymentDirect'"
+          color="success"
+          @click="submitConfig"
+          :disabled="!step2Valid"
+        >Submit</v-btn>
+        <v-btn flat @click="appConfigStep = 1">Back</v-btn>
+      </v-form>
     </v-stepper-content>
   </v-stepper>
 </template>
 
 <script>
 import { mapGetters } from "vuex";
+import {isValidJson} from "../utils/utils.js"
 
 export default {
   data() {
     return {
       appConfig: "",
       appConfigStep: 1,
+      step1Valid: false,
+      step2Valid: false,
+      showPw: false,
       commonServices: [
         { text: "Common Messaging Service", value: "cmsg" },
         { text: "Document Management Service", value: "dms" },
         { text: "Document Generation Service", value: "dgen", disabled: true }
       ],
-      userAppCfg: this.$store.state.userAppCfg
+      userAppCfg: this.$store.state.userAppCfg,
+      applicationAcronymRules: [
+        v => !!v || "Acroynm is required",
+        v => v.length <= 30 || "Acroynm must be 30 characters or less",
+        v =>
+          /^(?:[A-Z]{2,}[_]?)+[A-Z]{2,}$/g.test(v) ||
+          "Incorrect format. Hover the ? for details."
+      ],
+      applicationNameRules: [
+        v => !!v || "Name is required",
+        v => v.length <= 30 || "Name must be 30 characters or <b>less</b>"
```

Comment:
Wondering if we should be making the counter a variable instead of a magic number across the codebase